app/src/main/java/org/mozilla/focus/utils/TLDExtractor.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.utils
+
+import android.content.Context
+import kotlinx.coroutines.experimental.runBlocking
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.io.Reader
+import java.util.ArrayList
+import java.util.Collections
+import java.util.Comparator
+import java.util.regex.Pattern
+
+/**
+ * Finds the Top Level (or Secondary Level) Domain using a public list of TLD domains: http://publicsuffix.org/list/
+ * Initialize as soon as possible in application.
+ * Based on:
+ *  http://www.supermind.org/blog/1078/extracting-second-level-domains-and-top-level-domains-tld-from-a-url-in-java
+ **/
+object TLDExtractor {
+    private val stringBuilder: StringBuilder = StringBuilder()
+    private var pattern: Pattern? = null
+
+    @JvmStatic
+    fun init(context: Context) {
+        try {
+            val terms = ArrayList<String>()
+
+            val assetManager = context.assets
+            val inputStream: InputStream = assetManager.open("effective_tld_names.dat")
+            val inputStreamReader = InputStreamReader(inputStream)
+            val bufferedReader = BufferedReader(inputStreamReader as Reader)
+
+            runBlocking {
+                while (true) {
+                    var s = bufferedReader.readLine() ?: break
+                    runBlocking {
+                        s = s.trim { it <= ' ' }
+                        if (!s.isEmpty() && !s.startsWith("//") && !s.startsWith("!")) {
+                            terms.add(s)
+                        }
+                    }
+                }
+            }
+            Collections.sort(terms, StringLengthComparator)
+            for (t in terms) add(t)
+            compile()
+            bufferedReader.close()
+            inputStream.close()
+            inputStreamReader.close()
+        } catch (e: IOException) {
+            throw IllegalStateException(e)
+        }
```

Comment:
Crashing the app if we can't read the list is quite aggresive. I wonder if we want to continue without the list instead.